Feat s7 school admin profile page

joshua-hs opened the pull request:
This PR pertains to screen [S7 on the Figma board](https://www.figma.com/file/CkrBbFX2QQqIv0jdjOd4IR/Donate-to-Educate-%2F-Prototype?node-id=7218%3A188416&mode=dev) and allows users to edit the "About us" section within the school admin page, as such:

![image](https://github.com/CapgeminiInventUK/donate-to-educate-frontend/assets/10779091/9859a0bc-980f-48e9-9557-0abef5f50584)


To test the changes locally, spin up the front-end and navigate to http://localhost:5173/admin-dashboard/school.

I have not yet implemented functionality to make real API calls to retrieve or save data. I believe this was out of the scope of this ticket and will be tackled at a later date?

joshua-hs committed:
wip

joshua-hs committed:
feat: make About us section editable within school admin page

rs1257 commented on src/components/AdminActionTile/AdminActionTile.module.scss:
```diff
@@ -0,0 +1,64 @@
+  padding: 2.5rem;
+  gap: 2.5rem;
+  border-radius: 0.625rem;
+  border: 2px solid #e5e6eb;
```
Should move the colour to globals if not already

rs1257 commented on src/components/AdminActionTile/AdminActionTile.module.scss:
```diff
@@ -0,0 +1,64 @@
+  gap: 2.5rem;
+  border-radius: 0.625rem;
+  border: 2px solid #e5e6eb;
+  background: var(--primary-FEFCFD, #fefcfd);
```
Should use colour from globals

rs1257 commented on src/components/AdminActionTile/AdminActionTile.module.scss:
```diff
@@ -0,0 +1,64 @@
+  align-self: stretch;
+
+  h3 {
+    color: var(--primary-11356F, #11356f);
```
Same here

JakeReadman replied:
Also, a lot of these styles declared are unnecessary as they are set in globals, all unnecessary ones are:
`font-family`
`font-size`
`font-style`
and probs don't need 
`leading-trim`
`text-edge`
`line-height`
`letter-spacing`

rs1257 commented on src/components/AdminActionTile/AdminActionTile.module.scss:
```diff
@@ -0,0 +1,64 @@
+}
+
+.addSectionButtonText {
+  color: var(--primary-FEFCFD, #fefcfd);
```
And here

rs1257 commented on src/components/AdminActionTile/AdminActionTile.module.scss:
```diff
@@ -0,0 +1,64 @@
+  text-edge: cap;
+
+  /* H3 Heading 24px */
+  font-family: Poppins;
```
h3 already defined in globals

JakeReadman replied:
Aha, I should really read all Ryan's comments before I comment

rs1257 commented on src/components/EditableInformationTile/EditableInformationTile.module.scss:
```diff
@@ -0,0 +1,80 @@
+  gap: 2.5rem;
+  border-radius: 0.625rem;
+  border: 2px solid #e5e6eb;
+  background: var(--primary-FEFCFD, #fefcfd);
```
Same here, look like something figma has generated.

rs1257 commented on src/components/EditableInformationTile/EditableInformationTile.module.scss:
```diff
@@ -0,0 +1,80 @@
+}
+
+.addSectionButtonText {
+  color: var(--primary-FEFCFD, #fefcfd);
```
And here

rs1257 commented on src/components/EditableInformationTile/EditableInformationTile.module.scss:
```diff
@@ -0,0 +1,80 @@
+  align-items: center;
+  gap: 0.625rem;
+  border-radius: 0.125rem;
+  box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.04);
```
Is thos box shadow in globals?

rs1257 commented on src/components/EditableInformationTile/EditableInformationTile.module.scss:
```diff
@@ -0,0 +1,80 @@
+  color: #fff;
+  text-align: center;
+  leading-trim: both;
+  text-edge: cap;
```
Not convinced some of these related to trim and text edge have any effect worth checking in browser console.

rs1257 commented on src/components/InformationTile/InformationTile.module.scss:
```diff
@@ -0,0 +1,33 @@
+  align-items: flex-start;
+  align-self: stretch;
+  border-radius: 0.625rem;
+  background: #f1f1f1;
```
Move to globals if not already

rs1257 commented on src/components/InformationTile/InformationTile.module.scss:
```diff
@@ -0,0 +1,33 @@
+}
+
+.heading {
+  color: var(--primary-11356F, #11356f);
```
same

rs1257 commented on src/components/InformationTile/InformationTile.module.scss:
```diff
@@ -0,0 +1,33 @@
+}
+
+.subtext {
+  color: var(--primary-11356F, #11356f);
```
same

rs1257 commented on src/components/InformationTile/InformationTile.module.scss:
```diff
@@ -0,0 +1,33 @@
+  color: var(--primary-11356F, #11356f);
+
+  /* H2 Heading 32px */
+  font-family: Poppins;
```
go through all headers but I think they are all defined in globals already

rs1257 commented on src/components/SchoolBanner/SchoolBanner.module.scss:
```diff
@@ -39,11 +39,57 @@ h1 {
+}
+
+.italicized {
+  color: var(--primary-FEFCFD, #fefcfd);
```
same

rs1257 commented on src/pages/School/Admin/SchoolAdminDashboard.tsx:
```diff
@@ -0,0 +1,77 @@
+  }
+
+  // TODO - Make POST request to API to save new "About us" when user clicks Save button
+  function saveAboutUs(): void {
```
Can we use arrow notation? Same with function above.

JakeReadman commented on src/components/CancelButton/CancelButton.module.scss:
```diff
@@ -0,0 +1,21 @@
+.cancel {
+  color: #386eb3;
```
Colour in globals

JakeReadman commented on src/components/CancelButton/CancelButton.module.scss:
```diff
@@ -0,0 +1,21 @@
+.cancel {
+  color: #386eb3;
+  font-family: Poppins;
```
Again, a lot of these styles aren't necessary to declare here as done on a global level

JakeReadman commented on src/components/EditableInformationTile/EditableInformationTile.module.scss:
```diff
@@ -0,0 +1,80 @@
+  text-edge: cap;
+
+  /* H3 Heading 24px */
+  font-family: Poppins;
```
Again, h3 stuff already declared in globals

JakeReadman commented on src/components/EditableInformationTile/EditableInformationTile.module.scss:
```diff
@@ -0,0 +1,80 @@
+  leading-trim: both;
+  text-edge: cap;
+
+  /* H3 Heading 24px */
```
Remove

joshua-hs committed:
remove redundant css and use global css variables

joshua-hs committed:
Merge branch 'develop' into feat-s7-school-admin-profile-page

joshua-hs commented:
Hey @rs1257 and @JakeReadman, thanks for all your comments. My bad for all the css shenanigans, I just copied and pasted from figma dev mode and didn't realise there was a global variable file. I originally wrote replies to all your comments but accidentally refreshed the page before submitting any of them, so they all disappeared 😅 

tldr is I removed a lot of redundant css and am using the global file where appropriate. Think I addressed all your comments, but lemme know if you spot anything else that needs to change.

rs1257 commented on src/components/EditableInformationTile/EditableInformationTile.module.scss:
```diff
@@ -0,0 +1,59 @@
+  flex-direction: column;
+  gap: 2.5rem;
+  border-radius: 0.625rem;
+  border: 2px solid #e5e6eb;
```
colour here

Commit:
feat: add action buttons, move colour to globals, fix some merge errors

Commit:
feat: fix build issues

rs1257 commented:
Merging as we need to get the change in to unblock other work.
